src/stm32f0xx/gpio.c

```diff
@@ -1,62 +1,124 @@
 #include "gpio.h"
 
+#include <stdbool.h>
+#include <stdint.h>
+
 #include "stm32f0xx.h"
 #include "stm32f0xx_gpio.h"
+#include "stm32f0xx_rcc.h"
 
 static GPIO_TypeDef *gpio_port_map[] = {GPIOA, GPIOB, GPIOC, GPIOD, GPIOE, GPIOF};
+static uint32_t gpio_rcc_ahb_timer_map[] = {RCC_AHBPeriph_GPIOA, RCC_AHBPeriph_GPIOB,
+                                            RCC_AHBPeriph_GPIOC, RCC_AHBPeriph_GPIOD,
+                                            RCC_AHBPeriph_GPIOE, RCC_AHBPeriph_GPIOF};
+
+#define NUM_PINS 16
```

Comment:
I would recommend moving this to the header and prefixing it with the module name.